src/libraries/Microsoft.PowerFx.Core/Binding/DelegationWarningProvider.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using Microsoft.PowerFx.Core.App.ErrorContainers;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Syntax;
+
+namespace Microsoft.PowerFx.Core.Binding
+{
+    /// <summary>
+    /// An abstract class that may be implemented and used during binding to provide
+    /// warnings for specific delegation scenarios.  Visitation / provider methods are called
+    /// with a container pertaining to the formula that is being bound.
+    /// </summary>
+    internal abstract class DelegationHintProvider
+    {
+        public virtual bool TryGetWarning(BinaryOpNode node, out ErrorResourceKey? warningText)
+        {
+            warningText = null;
+            return false;
+        }
+        
+        public virtual bool TryGetWarning(CallNode node, TexlFunction function, out ErrorResourceKey? warningText)
```

Comment:
>TexlFunction function [](http://example.com/codeflow?start=57&length=21)

We don't really need to pass the function (it can be retrieved from the node), so we can make the interface simpler

Reply:
Nevermind, we don't necessarily have a binding in the provider.